src/advanced_inv.cpp

```diff
@@ -99,6 +91,38 @@ advanced_inventory::~advanced_inventory()
     }
 }
 
+void advanced_inventory::save_settings(bool only_panes)
+{
+    if(only_panes == false) {
+        uistate.adv_inv_last_coords = g->u.pos();
+        uistate.adv_inv_src = src;
+        uistate.adv_inv_dest = dest;
+    }
+    for(int i = 0; i < NUM_PANES; ++i) {
+        uistate.adv_inv_in_vehicle[i] = panes[i].in_vehicle();
+        uistate.adv_inv_area[i] = panes[i].get_area();
+        uistate.adv_inv_index[i] = panes[i].index;
+        uistate.adv_inv_filter[i] = panes[i].filter;
+    }
+}
+
+void advanced_inventory::load_settings()
+{
+    bool moved = uistate.adv_inv_last_coords != g->u.pos();
+    static const std::array<aim_location, NUM_PANES> default_areas = {
+        {AIM_ALL, AIM_INVENTORY}
+    };
+    for(int i = 0; i < NUM_PANES; ++i) {
+        auto location = (moved == true) ? default_areas[i] : 
```

Comment:
I'm not against '!', but it seems I've been told clarity is more important. ;-) I use AIM the same way, so i'll look at this. :-)